scripts/signers_framework/signers_analysis.py

```diff
@@ -0,0 +1,102 @@
+from datetime import datetime, timezone
+import requests
+import pandas as pd
+from rich.console import Console
+from rich.pretty import pprint
+
+from brownie import Contract, web3
+from helpers.addresses import r
+
+console = Console()
+
+# NOTE: request info and fill it up over time depending on rotations
+replacements = {
+    "techops_multisig": {
+        "0x10C7a2ca16116E5C998Bfa3BC9CEF464475B18ff": "0x0a9af7FAba0d5DF7A8C881e1B9cd679ee07Af8A2",
+        "0xeE8b29AA52dD5fF2559da2C50b1887ADee257556": "0xfA5bb45895Cb3C0aE5B1583Fe068f009A48F0187",
+        "0xBB2281cA5B4d07263112604D1F182AD0Ab26a252": "0xE8eA1D8B3a5A4CEC7E94AE330fF18E82B5D22fA6",
+    }
+}
+
+# we introduce times of judgement as args (default: Q2)
+# likely worth running once at the end of each quarter
+def main(start_time=1648771200, end_time=1656547200):
+    msigs = r.badger_wallets
+
+    console.print(
+        f"Time interval inspected: [{datetime.fromtimestamp(start_time, tz=timezone.utc)}, {datetime.fromtimestamp(end_time, tz=timezone.utc)}]"
+    )
+
+    for key, addr in msigs.items():
+        if "multisig" in key:
+            console.print(f"[green] Inspecting {key} at {addr}[/green]")
+
+            safe = Contract(addr)
+
+            total_tx = 0
+
+            # judging signed amount
+            owners_signed = dict.fromkeys(safe.getOwners(), 0)
```

Comment:
getting an error here on `.getOwners()`, even with fetching source again from explorer:
```
 Inspecting test_multisig_1 at 0x55949F769d0aF7453881435612561D109fFF07B8
Fetching source of 0x55949F769d0aF7453881435612561D109fFF07B8 from api.etherscan.io...
  File "brownie/_cli/run.py", line 50, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/signers_framework/signers_analysis.py", line 39, in main
    owners_signed = dict.fromkeys(safe.getOwners(), 0)
  File "brownie/network/contract.py", line 753, in __getattribute__
    raise AttributeError(f"Contract '{self._name}' object has no attribute '{name}'")
AttributeError: Contract 'Proxy' object has no attribute 'getOwners'
```

Reply:
it may be something local cause 1.1.1 safes should have that method. [ref](https://etherscan.io/address/0x34cfac646f301356faa8b21e94227e3583fe3f5f#code#L447)

Reply:
<img width="1321" alt="Screenshot 2022-08-16 at 16 30 56" src="https://user-images.githubusercontent.com/84875062/184905715-68e6e03f-93a2-46d4-ab77-31b096cfb7f4.png">

Reply:
turns out the contract wasnt registered as a proxy yet on etherscan, making `Contract.from_explorer` not look for an actual underlying implementation contract.

after letting etherscan know it is a proxy, it now returns the implementation contract.